cmake/SetupCharmAlgorithm.py

```diff
@@ -48,40 +49,25 @@ def create_interface_file(args):
               "\n" \
               "    entry void perform_algorithm(bool);\n" \
                             "\n" \
-              "    entry void start_phase(TYPENAME_PHASE_TYPE);\n" \
+              "    entry void start_phase(\n" \
+              "         typename ParallelComponent::metavariables::Phase);\n" \
               "\n" % (args['algorithm_name'], args['algorithm_name'])
 
-    if (args['algorithm_type'] == "nodegroup"):
-        ci_str += \
-            "    template <typename Action, typenameLDOTLDOTLDOT Args>\n" \
+    if args['algorithm_type'] == "nodegroup":
+        ci_str += "    template <typename Action, typename... Args>\n" \
             "    entry void threaded_action(\n" \
-            "               std::tuple<COMPUTE_VARIADIC_ARGS>& args);\n" \
+            "               std::tuple<Args...>& args);\n" \
             "\n" \
             "    template <typename Action>\n" \
             "    entry void threaded_action();\n" \
             "\n"
 
-    # A bug in Charm++ prevents entry methods with default argument values.
-    # The workaround is to have two entry methods and default the argument in
-    # the AlgorithmImpl member function.
-    if (args['algorithm_type'] == "group"
-            or args['algorithm_type'] == "nodegroup"):
-        ci_str += \
-            "    template <typename ReceiveTag, typename ReceiveData_t>\n" \
-            "    entry void receive_data(ReceiveTag_temporal_id&,\n" \
-            "                            ReceiveData_t&,\n" \
-            "                            bool enable_if_disabled);\n"
-        ci_str += \
-            "    template <typename ReceiveTag, typename ReceiveData_t>\n" \
-            "    entry void receive_data(ReceiveTag_temporal_id&,\n" \
-            "                            ReceiveData_t&);\n"
-    else:
-        ci_str += \
-            "    template <typename ReceiveTag, typename ReceiveData_t>\n" \
-            "    entry void receive_data(\n" \
-            "                            ReceiveTag_temporal_id&,\n" \
-            "                            ReceiveData_t&,\n" \
-            "                            bool enable_if_disabled = false);\n"
+    ci_str += \
+        "    template <typename ReceiveTag, typename ReceiveData_t>\n" \
+        "    entry void receive_data(\n" \
+        "                            typename ReceiveTag::temporal_id&,\n" \
+        "                            ReceiveData_t&,\n" \
+        "                            bool enable_if_disabled = false);\n"
```

Comment:
@kidder This change is related to the code and documentation here: https://github.com/sxs-collaboration/spectre/blob/1a830e7cd09bc3bfb969b0faa2bae7b3bdea099e/src/Parallel/Invoke.hpp#L44
Could you update that code or its documentation, or open an issue so we don't forget that the note in the documentation doesn't apply anymore and should be updated?